lib/modules/manager/terraform/hcl/index.ts

```diff
@@ -6,12 +6,236 @@ export async function parseHCL(
   fileName: string,
 ): Promise<TerraformDefinitionFile | null> {
   try {
-    return await parse(fileName, content);
+    if (fileName.endsWith('.tf') || fileName.endsWith('.hcl')) {
+      return await parse(fileName, content);
+    } else if (
+      fileName.endsWith('.tf.json') ||
+      fileName.endsWith('.hcl.json')
+    ) {
+      return await parse(fileName, terraformJsonToHcl(JSON.parse(content)));
+    } else {
+      return null;
+    }
   } catch /* istanbul ignore next */ {
     return null;
   }
 }
 
-export function parseJSON(content: string): TerraformDefinitionFile | null {
-  return JSON.parse(content);
+// take a JS value and return valid HCL literal)
+function toHclLiteral(value: any): string {
+  if (typeof value === 'string') {
+    return `"${value}"`;
+  }
+  if (typeof value === 'number' || typeof value === 'boolean') {
+    return String(value);
+  }
+  if (Array.isArray(value)) {
+    const elems = value.map((v) => toHclLiteral(v)).join(', ');
+    return `[ ${elems} ]`;
+  }
+  if (value === null) {
+    return 'null';
+  }
+  if (typeof value === 'object') {
+    let out = '{\n';
+    for (const [k, v] of Object.entries(value)) {
+      out += `  ${k} = ${toHclLiteral(v)}\n`;
+    }
+    out += '}';
+    return out;
+  }
+  return `"${String(value)}"`;
+}
+
+// Safely normalize a block that might be either an array or a single object
+function ensureArray<T>(maybeArray: T | T[] | undefined): T[] {
+  if (Array.isArray(maybeArray)) {
+    return maybeArray;
+  }
+  if (maybeArray === undefined || maybeArray === null) {
+    return [];
+  }
+  return [maybeArray];
+}
+
+function terraformJsonToHcl(terraformJson: any): string {
```

Comment:
Put this and its helper fns in a util file

Reply:
moved into a util file